src/software/sensor_fusion/filter/robot_filter.cpp

```diff
@@ -8,9 +8,11 @@ RobotFilter::RobotFilter(Robot current_robot_state, Duration expiry_buffer_durat
 
 RobotFilter::RobotFilter(RobotDetection current_robot_state,
                          Duration expiry_buffer_duration)
-    : current_robot_state(current_robot_state.id, current_robot_state.position,
-                          Vector(0, 0), current_robot_state.orientation,
-                          AngularVelocity::zero(), current_robot_state.timestamp),
+    : current_robot_state(
+          current_robot_state.id,
+          TimestampedRobotState(current_robot_state.position, Vector(0, 0),
+                                current_robot_state.orientation, AngularVelocity::zero(),
+                                current_robot_state.timestamp)),
```

Comment:
I'm a bit confused about what this change is about? Likewise for the similar change below.

Reply:
not sure either, reverted